File: articles/decision-making-under-deep-uncertainty/r/run_all_deep_uncertainty_workflows.R
# run_all_deep_uncertainty_workflows.R
# Run all R workflows for Decision-Making Under Deep Uncertainty.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_making_under_deep_uncertainty_workflow.R",
  "ambiguity_profile_tables.R",
  "regret_tables.R",
  "vulnerability_tables.R",
  "adaptive_strategy_tables.R",
  "deep_uncertainty_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R deep uncertainty workflows completed.")

File: articles/decision-making-under-deep-uncertainty/python/run_all_deep_uncertainty_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision-Making Under Deep Uncertainty."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "ambiguity_profile_model.py",
    "exploratory_scenario_scan.py",
    "regret_analysis.py",
    "threshold_compliance.py",
    "vulnerability_analysis.py",
    "adaptive_strategy_simulation.py",
    "decision_record_exporter.py",
    "decision_making_under_deep_uncertainty_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python deep uncertainty workflows completed.")


if __name__ == "__main__":
    main()
